installer/tue-install-impl.bash

```diff
@@ -270,20 +271,111 @@ function tue-install-git
     tue-install-debug "Desired version: $version"
     if [ -n "$version" ];
     then
-        _try_branch "$targetdir" "$version"
+        _try_branch_git "$targetdir" "$version"
         res="$res $_try_branch_res"
     fi
 
     tue-install-debug "Desired branch: $BRANCH"
     if [ -n "$BRANCH" ]; #Cannot be combined with version-if because this one might not exist
     then
-        _try_branch "$targetdir" "$BRANCH"
+        _try_branch_git "$targetdir" "$BRANCH"
         res="$res $_try_branch_res"
     fi
 
     _show_update_message "$TUE_INSTALL_CURRENT_TARGET" "$res"
 }
 
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+function tue-install-hg
+{
+    tue-install-debug "tue-install-hg $*"
+
+    local repo=$1
+    local targetdir=$2
+    local version=$3
+
+    if [ ! -d "$targetdir" ]
+    then
+        tue-install-debug "hg clone $repo $targetdir"
+        res=$(hg clone "$repo" "$targetdir" 2>&1)
+        TUE_INSTALL_HG_PULL_Q+=$targetdir
+    else
+        # Check if we have already pulled the repo
+        if [[ $TUE_INSTALL_HG_PULL_Q =~ $targetdir ]]
+        then
+            tue-install-debug "Repo previously pulled, skipping"
+            # We have already pulled this repo, skip it
+            res=
+        else
+            # Switch url of origin to use https/ssh if different
+            # Get current remote url
+            local current_url
+            current_url=$(hg -R "$targetdir" cfg paths.default | awk '{print $2}')
```

Comment:
I think there is a typo here. `cfg` -> `config`

Reply:
nope, `hgcfg` exposes `cfg`. I prefer to use `cfg`, so I am sure the `hgcfg` extension is used. (And error throws if there is something wrong with `hgcfg`) Which has a slightly different output.